dfastmi/batch/AnalyserAndReporterWaqua.py

```diff
@@ -96,180 +86,282 @@ def analyse_and_report_waqua(
+        
+    def analyse(self, fraction_of_year : Vector, rsigma : Vector) -> tuple[bool, OutputDataWaqua]:
+        """
+        Read data from samples files exported from WAQUA simulations and performanalysis.
 
-    Returns
-    -------
-    dzq : numpy.ndarray
-        Array containing equilibrium bed level change.
-    firstm : int
-        Minimum M index read (0 if reduced_output is False).
-    firstn : int
-        Minimum N index read (0 if reduced_output is False).
-    """
-    cblok = str(stage)
-    if display:
-        ApplicationSettingsHelper.log_text("input_xyz", dict={"stage": stage, "q": q})
-        ApplicationSettingsHelper.log_text("---")
-        ApplicationSettingsHelper.log_text("")
-
-    discriptions = ApplicationSettingsHelper.get_text("file_descriptions")
-    quantities = ["velocity-zeta.001", "waterdepth-zeta.001", "velocity-zeta.002"]
-    files = []
-    for i in range(3):
-        if display:
-            ApplicationSettingsHelper.log_text("input_xyz_name", dict={"name": discriptions[i]})
-        cifil = "xyz_" + quantities[i] + ".Q" + cblok + ".xyz"
-        if display:
-            print(cifil)
-        if not os.path.isfile(cifil):
-            ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil})
-            ApplicationSettingsHelper.log_text("file_not_found", dict={"name": cifil}, file=report)
-            ApplicationSettingsHelper.log_text("end_program", file=report)
+        Arguments
+        ---------
+        fraction_of_year : Vector
+            Fraction of year represented by each forcing condition.
+        rsigma : Vector
+            Array of relaxation factors; one per forcing condition.
+
+        Returns
+        -------
+        success : bool
+            Flag indicating whether analysis could be carried out.
+        output_data : OutputDataWaqua
+            data used to generate a report.
+        """   
+        success, dzq, firstm, firstn = self._get_output_values()
+        output_data = self._calculate_output_data(fraction_of_year, rsigma, dzq, firstm, firstn)
+        return success, output_data
+
+    def _get_output_values(self) -> Tuple[bool, numpy.ndarray, int, int]:
+        success = True
+        first_discharge = True
+        dzq : List[Optional[Union[float, numpy.ndarray]]]
+        dzq = [None] * len(self.discharges)
+        for i in range(3):
+            if success and self.apply_q[i]:
+                if first_discharge:
+                    dzq[i], firstm, firstn = self._get_values_waqua(i+1, self.discharges[i])
+                    first_discharge = False
+                else:
+                    dzq[i], _, _ = self._get_values_waqua(i+1, self.discharges[i])
+                if dzq[i] is None:
+                    success = False
+            else:
+                dzq[i] = 0.0
+        return success, dzq, firstm, firstn
+
+    def _get_values_waqua(self, stage: int, discharge_value: float) -> Tuple[numpy.ndarray, int, int]:
```

Comment:
Should this not be 3 methods?

Reply:
moved into seperate methods to just return a single variable.